BinariesAPI.md

```diff
@@ -0,0 +1,107 @@
+# JULEP BinariesAPI
+
+- **Title:** A structured HTTP API for getting Julia binaries
+- **Authors:** Eric Davies <<iamed2@gmail.com>>
+- **Created:** June 16, 2017
+- **Status:** proposed
+
+## Abstract
+
+The URLs for fetching Julia binaries of various versions have changed a few times across
+releases.
+The latest change necessitated PRs to multiple repos to change AppVeyor scripts.
+This Julep proposes a level of abstraction over the Julia binary URLs which is decoupled
+from the storage service used for the binaries.
```

Comment:
What's the value in adding a third? If we're going to pick a url structure and naming convention that we intend to keep stable forever, the caching layer can be laid out that way.